cpp/src/strings/convert/is_valid_fixed_point.cu

```diff
@@ -0,0 +1,173 @@
+#include <cudf/strings/detail/utilities.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <strings/utilities.cuh>
+#include <strings/utilities.hpp>
+
+#include <thrust/logical.h>
+
+namespace cudf {
+namespace strings {
+
+/**
+   * Parses this UTF8String(trimmed if needed) to INT8/16/32/64...
+   *
+   * Note that, in this method we accumulate the result in negative format, and convert it to
+   * positive format at the end, if this string is not started with '-'. This is because min value
+   * is bigger than max value in digits, e.g. Long.MAX_VALUE is '9223372036854775807' and
+   * Long.MIN_VALUE is '-9223372036854775808'.
+   *
+   * This code is mostly copied from LazyLong.parseLong in Hive.
+   *
+ * @param d_str String to check.
+ * @param allow_decimal Decimal format or not
+ * @param min_value min_value that corresponds to the type that is checking
+ * @return true if string has valid integer characters
+   */
+__device__ bool is_valid_fixed_point(string_view const& d_str, bool allow_decimal, long min_value)
+{
+  int offset = 0;
+  size_type bytes = d_str.size_bytes();
+  const char* data    = d_str.data();
+  while (offset < bytes && data[offset] == ' ') ++offset;
+  if (offset == bytes)  return false;
+
+  int end = bytes - 1;
+  while (end > offset && data[end] == ' ') --end;
+
+  char c_sign = data[offset];
+  const bool negative = c_sign == '-';
+  if (negative || c_sign == '+'){
+    if (end - offset == 0)  return false;
+    ++offset;
+  }
+
+  const char separator = '.';
+  const int radix = 10;
+  const long stop_value = min_value / radix;
+  long result = 0;
+
+  while (offset <= end) {
+    const char c = data[offset];
+    ++offset;
+    // We allow decimals and will return a truncated integral in that case.
+    // Therefore we won't throw an exception here (checking the fractional
+    // part happens below.
+    if (c == separator && allow_decimal)  break;
+
+    int digit;
+    if (c >= '0' && c <= '9'){
+      digit = c - '0';
+    } else {
+      return false;
+    }
+
+    // We are going to process the new digit and accumulate the result. However, before doing
+    // this, if the result is already smaller than the stopValue(Long.MIN_VALUE / radix), then
+    // result * 10 will definitely be smaller than minValue, and we can stop.
+    if (result < stop_value)  return false;
+
+    result = result * radix - digit;
+
+    // Since the previous result is less than or equal to stopValue(Long.MIN_VALUE / radix), we
+    // can just use `result > 0` to check overflow. If result overflows, we should stop.
+    if (result > 0) return false;
+  }
+  // This is the case when we've encountered a decimal separator. The fractional
+  // part will not change the number, but we will verify that the fractional part
+  // is well formed.
+  while (offset <= end) {
```

Comment:
Why are we doing this?

Reply:
in order to verify the fractional part whether is well formed. if allow_decimal is true and has character '.' , it will break from while loop , so here we need to verify the fractional part after the decimal point.